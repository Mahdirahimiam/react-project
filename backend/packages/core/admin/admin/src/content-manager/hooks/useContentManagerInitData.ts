import { useEffect, useRef } from 'react';

import { useNotifyAT } from '@strapi/design-system';
import {
  Permission,
  hasPermissions,
  useFetchClient,
  useNotification,
  useRBACProvider,
  useStrapiApp,
} from '@strapi/helper-plugin';
import { Contracts } from '@strapi/plugin-content-manager/_internal/shared';
import { stringify } from 'qs';
import { useIntl } from 'react-intl';

import { HOOKS } from '../../constants';
import { useTypedDispatch, useTypedSelector } from '../../core/store/hooks';
import { getTranslation } from '../utils/translations';

const { MUTATE_COLLECTION_TYPES_LINKS, MUTATE_SINGLE_TYPES_LINKS } = HOOKS;

interface ContentManagerLink {
  permissions: Permission[];
  search: string | null;
  kind: string;
  title: string;
  to: string;
  uid: string;
  name: string;
  isDisplayed: boolean;
}

/**
 * TODO: refactor this whole thing.
 */
const useContentManagerInitData = () => {
  const dispatch = useTypedDispatch();
  const toggleNotification = useNotification();
  const state = useTypedSelector((state) => state['content-manager_app']);

  const { allPermissions } = useRBACProvider();
  const { runHookWaterfall } = useStrapiApp();
  const { notifyStatus } = useNotifyAT();
  const { formatMessage } = useIntl();
  const { get } = useFetchClient();

  const fetchDataRef = useRef(async () => {
    dispatch({
      type: 'ContentManager/App/GET_INIT_DATA',
    });

    try {
      const {
        data: {
          data: { components, contentTypes: models, fieldSizes },
        },
      } = await get<Contracts.Init.GetInitData.Response>('/content-manager/init');
      notifyStatus(
        formatMessage({
          id: getTranslation('App.schemas.data-loaded'),
          defaultMessage: 'The schemas have been successfully loaded.',
        })
      );

      const {
        data: { data: contentTypeConfigurations },
      } = await get<Contracts.ContentTypes.FindContentTypesSettings.Response>(
        '/content-manager/content-types-settings'
      );

      /**
       * We group these by the two types we support. We do with an object because we can use default
       * values of arrays to make sure we always have an array to manipulate further on if, for example,
       * a user has not made any single types.
       *
       * This means we have to manually add new content types to this hook if we add a new type â€“ but
       * the safety is worth it.
       */
      const { collectionType: collectionTypeLinks, singleType: singleTypeLinks } = models.reduce<{
        collectionType: Contracts.ContentTypes.ContentType[];
        singleType: Contracts.ContentTypes.ContentType[];
      }>(
        (acc, model) => {
          acc[model.kind].push(model);
          return acc;
        },
        {
          collectionType: [],
          singleType: [],
        }
      );

      const collectionTypeSectionLinks = generateLinks(
        collectionTypeLinks,
        'collectionTypes',
        contentTypeConfigurations
      );

      const singleTypeSectionLinks = generateLinks(singleTypeLinks, 'singleTypes');

      // Collection Types verifications
      const collectionTypeLinksPermissions = await Promise.all(
        collectionTypeSectionLinks.map(({ permissions }) =>
          hasPermissions(allPermissions, permissions)
        )
      );
      const authorizedCollectionTypeLinks = collectionTypeSectionLinks.filter(
        (_, index) => collectionTypeLinksPermissions[index]
      );

      // Single Types verifications
      const singleTypeLinksPermissions = await Promise.all(
        singleTypeSectionLinks.map(({ permissions }) => hasPermissions(allPermissions, permissions))
      );
      const authorizedSingleTypeLinks = singleTypeSectionLinks.filter(
        (_, index) => singleTypeLinksPermissions[index]
      );

      const { ctLinks } = runHookWaterfall(MUTATE_COLLECTION_TYPES_LINKS, {
        ctLinks: authorizedCollectionTypeLinks,
        models,
      });
      const { stLinks } = runHookWaterfall(MUTATE_SINGLE_TYPES_LINKS, {
        stLinks: authorizedSingleTypeLinks,
        models,
      });

      dispatch({
        type: 'ContentManager/App/SET_INIT_DATA',
        data: {
          authorizedCollectionTypeLinks: ctLinks,
          authorizedSingleTypeLinks: stLinks,
          contentTypeSchemas: models,
          components,
          fieldSizes,
        },
      });
    } catch (err) {
      console.error(err);

      toggleNotification({ type: 'warning', message: { id: 'notification.error' } });
    }
  });

  useEffect(() => {
    fetchDataRef.current();

    return () => {
      dispatch({
        type: 'ContentManager/App/RESET_INIT_DATA',
      });
    };
  }, [dispatch, toggleNotification]);

  return { ...state, refetchData: fetchDataRef.current };
};

const generateLinks = (
  links: Contracts.ContentTypes.ContentType[],
  type: 'collectionTypes' | 'singleTypes',
  configurations: Contracts.ContentTypes.FindContentTypesSettings.Response['data'] = []
) => {
  return links
    .filter((link) => link.isDisplayed)
    .map((link) => {
      const collectionTypesPermissions = [
        { action: 'plugin::content-manager.explorer.create', subject: link.uid },
        { action: 'plugin::content-manager.explorer.read', subject: link.uid },
      ];
      const singleTypesPermissions = [
        { action: 'plugin::content-manager.explorer.read', subject: link.uid },
      ];
      const permissions =
        type === 'collectionTypes' ? collectionTypesPermissions : singleTypesPermissions;

      const currentContentTypeConfig = configurations.find(({ uid }) => uid === link.uid);

      let search = null;

      if (currentContentTypeConfig) {
        const searchParams = {
          page: 1,
          pageSize: currentContentTypeConfig.settings.pageSize,
          sort: `${currentContentTypeConfig.settings.defaultSortBy}:${currentContentTypeConfig.settings.defaultSortOrder}`,
        };

        search = stringify(searchParams, { encode: false });
      }

      return {
        permissions,
        search,
        kind: link.kind,
        title: link.info.displayName,
        to: `/content-manager/${
          link.kind === 'collectionType' ? 'collection-types' : 'single-types'
        }/${link.uid}`,
        uid: link.uid,
        // Used for the list item key in the helper plugin
        name: link.uid,
        isDisplayed: link.isDisplayed,
      } satisfies ContentManagerLink;
    });
};

export { useContentManagerInitData };
export type { ContentManagerLink };
