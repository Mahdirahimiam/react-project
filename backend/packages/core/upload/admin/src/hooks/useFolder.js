import { useFetchClient, useNotification } from '@strapi/helper-plugin';
import { useQuery } from 'react-query';

import pluginId from '../pluginId';
import { getTrad } from '../utils';

export const useFolder = (id, { enabled = true } = {}) => {
  const toggleNotification = useNotification();
  const { get } = useFetchClient();

  const { data, error, isLoading } = useQuery(
    [pluginId, 'folder', id],
    async () => {
      const {
        data: { data },
      } = await get(`/upload/folders/${id}`, {
        params: {
          populate: {
            parent: {
              populate: {
                parent: '*',
              },
            },
          },
        },
      });

      return data;
    },
    {
      retry: false,
      enabled,
      staleTime: 0,
      cacheTime: 0,
      onError() {
        toggleNotification({
          type: 'warning',
          message: {
            id: getTrad('notification.warning.404'),
            defaultMessage: 'Not found',
          },
        });
      },
    }
  );

  return { data, error, isLoading };
};
