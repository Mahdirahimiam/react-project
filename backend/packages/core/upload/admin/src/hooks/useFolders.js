import * as React from 'react';

import { useNotifyAT } from '@strapi/design-system';
import { useFetchClient, useNotification } from '@strapi/helper-plugin';
import { stringify } from 'qs';
import { useIntl } from 'react-intl';
import { useQuery } from 'react-query';

import pluginId from '../pluginId';

export const useFolders = ({ enabled = true, query = {} } = {}) => {
  const { formatMessage } = useIntl();
  const toggleNotification = useNotification();
  const { notifyStatus } = useNotifyAT();
  const { folder, _q, ...paramsExceptFolderAndQ } = query;
  const { get } = useFetchClient();

  let params;

  if (_q) {
    params = {
      ...paramsExceptFolderAndQ,
      pagination: {
        pageSize: -1,
      },
      _q,
    };
  } else {
    params = {
      ...paramsExceptFolderAndQ,
      pagination: {
        pageSize: -1,
      },
      filters: {
        $and: [
          ...(paramsExceptFolderAndQ?.filters?.$and ?? []),
          {
            parent: {
              id: folder ?? {
                $null: true,
              },
            },
          },
        ],
      },
    };
  }

  const { data, error, isLoading } = useQuery(
    [pluginId, 'folders', stringify(params)],
    async () => {
      const {
        data: { data },
      } = await get('/upload/folders', { params });

      return data;
    },
    {
      enabled,
      staleTime: 0,
      cacheTime: 0,
      onError() {
        toggleNotification({
          type: 'warning',
          message: { id: 'notification.error' },
        });
      },
    }
  );

  React.useEffect(() => {
    if (data) {
      notifyStatus(
        formatMessage({
          id: 'list.asset.at.finished',
          defaultMessage: 'The folders have finished loading.',
        })
      );
    }
  }, [data, formatMessage, notifyStatus]);

  return { data, error, isLoading };
};
