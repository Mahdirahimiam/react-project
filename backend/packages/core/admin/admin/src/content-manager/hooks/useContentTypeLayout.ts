import { useCallback } from 'react';

import { useTypedSelector } from '../../core/store/hooks';

const useContentTypeLayout = () => {
  const currentLayout = useTypedSelector(
    (state) => state['content-manager_editViewLayoutManager'].currentLayout
  );

  const getComponentLayout = useCallback(
    (componentUid: string) => {
      return currentLayout?.components?.[componentUid] ?? {};
    },
    [currentLayout]
  );

  return { ...currentLayout, getComponentLayout };
};

export { useContentTypeLayout };
