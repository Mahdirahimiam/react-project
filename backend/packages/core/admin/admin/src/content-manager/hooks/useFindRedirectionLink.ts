import { useQueryParams } from '@strapi/helper-plugin';
import QueryString, { parse, stringify } from 'qs';

import { useTypedSelector } from '../../core/store/hooks';
import { isObject } from '../../utils/objects';

import type { ContentManagerAppState } from '../pages/App';

const useFindRedirectionLink = (slug: string) => {
  const [{ rawQuery }] = useQueryParams();
  const collectionTypesMenuLinks = useTypedSelector(
    (state) => state['content-manager_app'].collectionTypeLinks
  );
  const redirectionLink = getRedirectionLink(collectionTypesMenuLinks, slug, rawQuery);

  return redirectionLink;
};

/**
 * Updates the leafs of the first argument
 */
const mergeParams = (
  initialParams: QueryString.ParsedQs,
  params: QueryString.ParsedQs
): QueryString.ParsedQs => {
  return Object.keys(initialParams).reduce<QueryString.ParsedQs>((acc, current) => {
    const initialValue = initialParams[current];
    const nextValue = params[current] ?? initialValue;

    if (isObject(initialValue)) {
      // @ts-expect-error â€“ TODO: fix this type error.
      return { ...acc, [current]: mergeParams(initialValue, nextValue) };
    }

    acc[current] = nextValue;

    return acc;
  }, {});
};

const getRedirectionLink = (
  links: ContentManagerAppState['collectionTypeLinks'],
  slug: string,
  rawQuery: string
) => {
  const matchingLink = links.find(({ to }) => to.includes(slug));

  if (!matchingLink) {
    return '/';
  }

  const { to, search } = matchingLink;
  const searchQueryParams = parse(search ?? '');
  const currentQueryParams = parse(rawQuery.substring(1));

  const mergedParams = mergeParams(searchQueryParams, currentQueryParams);

  const link = `${to}?${stringify(mergedParams, { encode: false })}`;

  return link;
};

export { useFindRedirectionLink };
