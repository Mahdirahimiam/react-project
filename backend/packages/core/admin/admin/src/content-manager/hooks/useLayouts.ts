import * as React from 'react';

import { useFetchClient } from '@strapi/helper-plugin';
import { Contracts } from '@strapi/plugin-content-manager/_internal/shared';
import { useQuery } from 'react-query';

import { useTypedSelector } from '../../core/store/hooks';
import { selectSchemas } from '../pages/App';
import { formatLayouts } from '../utils/layouts';

const useContentTypeLayout = (contentTypeUID: string) => {
  const schemas = useTypedSelector(selectSchemas);
  const { get } = useFetchClient();

  const { data, isLoading, refetch } = useQuery(
    ['content-manager', 'content-types', contentTypeUID, 'configuration'],
    async () => {
      const {
        data: { data },
      } = await get<Contracts.ContentTypes.FindContentTypeConfiguration.Response>(
        `/content-manager/content-types/${contentTypeUID}/configuration`
      );

      return data;
    }
  );

  const layout = React.useMemo(() => (data ? formatLayouts(data, schemas) : null), [data, schemas]);

  return {
    isLoading,
    layout,
    updateLayout: refetch,
  };
};

export { useContentTypeLayout };
