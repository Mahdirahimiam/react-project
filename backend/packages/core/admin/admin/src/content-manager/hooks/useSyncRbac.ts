import { useEffect } from 'react';

import { Permission } from '@strapi/helper-plugin';
import produce from 'immer';

import { useTypedDispatch, useTypedSelector } from '../../core/store/hooks';

import type { RBACState } from '../../components/RBACProvider';

/* -------------------------------------------------------------------------------------------------
 * useSyncRbac
 * -----------------------------------------------------------------------------------------------*/

const useSyncRbac = (
  query: { plugins?: object },
  collectionTypeUID: string,
  containerName = 'listView'
): {
  isValid: boolean;
  permissions: Permission[] | null;
} => {
  const dispatch = useTypedDispatch();

  const collectionTypesRelatedPermissions = useTypedSelector(
    (state) => state.rbacProvider.collectionTypesRelatedPermissions
  );
  const permissions = useTypedSelector((state) => state['content-manager_rbacManager'].permissions);

  const relatedPermissions = collectionTypesRelatedPermissions[collectionTypeUID];

  useEffect(() => {
    if (relatedPermissions) {
      dispatch({
        type: SET_PERMISSIONS,
        permissions: relatedPermissions,
        __meta__: {
          plugins: query ? query.plugins : null,
          containerName,
        },
      });

      return () => {
        dispatch({ type: RESET_PERMISSIONS });
      };
    }

    return () => {};
  }, [relatedPermissions, dispatch, query, containerName]);

  // Check if the permissions are related to the current collectionTypeUID
  const isPermissionMismatch =
    permissions?.some((permission) => permission.subject !== collectionTypeUID) ?? true;

  return {
    isValid: Boolean(permissions) && !isPermissionMismatch,
    permissions,
  };
};

/* -------------------------------------------------------------------------------------------------
 * Reducer
 * -----------------------------------------------------------------------------------------------*/

interface SyncRbacState {
  permissions: RBACState['collectionTypesRelatedPermissions'][string][string] | null;
}

const initialState = {
  permissions: null,
} satisfies SyncRbacState;

const SET_PERMISSIONS = 'ContentManager/RBACManager/SET_PERMISSIONS';
const RESET_PERMISSIONS = 'ContentManager/RBACManager/RESET_PERMISSIONS';

interface SetPermissionsAction {
  type: typeof SET_PERMISSIONS;
  permissions: RBACState['collectionTypesRelatedPermissions'][string];
  __meta__?: {
    plugins: object;
    containerName: string;
  };
}

interface ResetPermissionsAction {
  type: typeof RESET_PERMISSIONS;
}

type Action = SetPermissionsAction | ResetPermissionsAction;

const reducer = (state: SyncRbacState = initialState, action: Action) =>
  // eslint-disable-next-line consistent-return
  produce(state, (draftState) => {
    switch (action.type) {
      case SET_PERMISSIONS: {
        draftState.permissions = Object.entries(action.permissions).reduce<
          NonNullable<SyncRbacState['permissions']>
        >((acc, current) => {
          return [...acc, ...current[1]];
        }, []);
        break;
      }
      case RESET_PERMISSIONS: {
        draftState.permissions = null;
        break;
      }
      default:
        return draftState;
    }
  });

export { useSyncRbac, reducer };
export type { SyncRbacState, RBACState };
