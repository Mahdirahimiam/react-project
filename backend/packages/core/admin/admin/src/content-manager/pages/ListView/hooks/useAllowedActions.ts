import { useRBAC } from '@strapi/helper-plugin';

import { useTypedSelector } from '../../../../core/store/hooks';
import { generatePermissionsObject } from '../../../utils/permissions';

const useAllowedActions = (slug: string) => {
  const viewPermissions = generatePermissionsObject(slug);
  const permissions = useTypedSelector((state) => state['content-manager_rbacManager'].permissions);
  const { allowedActions } = useRBAC(viewPermissions, permissions ?? []);
  return allowedActions;
};

export { useAllowedActions };
