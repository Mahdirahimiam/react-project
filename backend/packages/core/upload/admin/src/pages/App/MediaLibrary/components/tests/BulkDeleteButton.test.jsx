import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { act, fireEvent, render, waitFor } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';

import { useBulkRemove } from '../../../../../hooks/useBulkRemove';
import { BulkDeleteButton } from '../BulkDeleteButton';

jest.mock('../../../../../hooks/useBulkRemove');

const setup = (
  props = {
    selected: [],
    onSuccess: jest.fn(),
  }
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        refetchOnWindowFocus: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={lightTheme}>
        <MemoryRouter>
          <IntlProvider locale="en">
            <BulkDeleteButton {...props} />
          </IntlProvider>
        </MemoryRouter>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

describe('BulkDeleteButton', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  test('opens confirm dialog before the API call', () => {
    const onSuccessSpy = jest.fn();
    const { getByText } = setup({
      onSuccess: onSuccessSpy,
      selected: [{ type: 'asset' }],
    });
    const removeSpy = jest.fn().mockResolvedValueOnce({});

    useBulkRemove.mockReturnValueOnce({
      isLoading: false,
      error: null,
      remove: removeSpy,
    });

    act(() => {
      fireEvent.click(getByText('Delete'));
    });

    expect(getByText('Are you sure you want to delete this?')).toBeInTheDocument();

    act(() => {
      fireEvent.click(getByText('Confirm'));
    });

    expect(removeSpy).toBeCalledWith([{ type: 'asset' }]);
    waitFor(() => expect(onSuccessSpy).toBeCalledTimes(1));
  });
});
