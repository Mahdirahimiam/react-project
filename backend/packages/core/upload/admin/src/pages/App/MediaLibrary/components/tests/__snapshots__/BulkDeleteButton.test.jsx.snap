// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkDeleteButton renders 1`] = `
.c6 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c5 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c0 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c2 {
  position: relative;
  outline: none;
}

.c2 > svg {
  height: 12px;
  width: 12px;
}

.c2 > svg > g,
.c2 > svg path {
  fill: #ffffff;
}

.c2[aria-disabled='true'] {
  pointer-events: none;
}

.c2:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c2:focus-visible {
  outline: none;
}

.c2:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c3 {
  height: 2rem;
  border: 1px solid #f5c0b8;
  background: #fcecea;
}

.c3 svg {
  height: 0.75rem;
  width: auto;
}

.c3[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c3[aria-disabled='true'] .c4 {
  color: #666687;
}

.c3[aria-disabled='true'] svg > g,
.c3[aria-disabled='true'] svg path {
  fill: #666687;
}

.c3[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c3[aria-disabled='true']:active .c4 {
  color: #666687;
}

.c3[aria-disabled='true']:active svg > g,
.c3[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c3:hover {
  background-color: #ffffff;
}

.c3:active {
  background-color: #ffffff;
  border: 1px solid #d02b20;
}

.c3:active .c4 {
  color: #d02b20;
}

.c3:active svg > g,
.c3:active svg path {
  fill: #d02b20;
}

.c3 .c4 {
  color: #b72b1a;
}

.c3 svg > g,
.c3 svg path {
  fill: #b72b1a;
}

<div>
  <button
    aria-disabled="false"
    class="c0 c1 c2 c3"
    type="button"
  >
    <div
      aria-hidden="true"
      class=""
    >
      <svg
        fill="none"
        height="1rem"
        viewBox="0 0 24 24"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.236 6.149a.2.2 0 0 0-.197.233L6 24h12l2.96-17.618a.2.2 0 0 0-.196-.233H3.236ZM21.8 1.983c.11 0 .2.09.2.2v1.584a.2.2 0 0 1-.2.2H2.2a.2.2 0 0 1-.2-.2V2.183c0-.11.09-.2.2-.2h5.511c.9 0 1.631-1.09 1.631-1.983h5.316c0 .894.73 1.983 1.631 1.983H21.8Z"
          fill="#32324D"
        />
      </svg>
    </div>
    <span
      class="c4 c5"
    >
      Delete
    </span>
  </button>
  <div
    class="c6"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
